### PR TITLE
Fix: Fixed issue where search results would sometimes use the Columns View

### LINKED ISSUE #13439: Bug: Issue with Search Page Layout Synchronization
### Description

I'm experiencing an issue with the search functionality on my computer. When I perform an initial search, the page displaying the items containing the desired keyword appears in detailed view. However, when I initiate a second search while still on the search page, the results of the second search are displayed correctly, but the layout is synchronized with the settings. In my case, for the second search, it displays in column view since I have synchronized this layout.

### Steps To Reproduce

1. Navigate to the settings, check "Sync layout and sorting preferences across directories", and make sure the column layout is selected in "Layout"
2. Perform a search from anywhere as long as you are in column mode.

### Requirements

Investigate why the search page layout is being synchronized with your settings.

### Files Version

2.5.29.0

### Windows Version

22H2 22621.1992

### Log File

...

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #13439...

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Set folder location to use Columns View
   2. Submit a search and confirm it uses the Details View to display the results
   3. Submit another search and confirm it's still using the Details View
   4. Go back to the original location and confirm it uses the Columns View

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

File: src/Files.App/Views/Shells/BaseShellPage.cs [hunk 1]
```diff
@@ -493,10 +493,12 @@ public void SubmitSearch(string query)
...
 			};
 
-			if (this is ColumnShellPage)
+			var layout = InstanceViewModel.FolderSettings.GetLayoutType(FilesystemViewModel.WorkingDirectory);
+
+			if (layout == typeof(ColumnsLayoutPage))
 				NavigateToPath(FilesystemViewModel.WorkingDirectory, typeof(DetailsLayoutPage), args);
 			else
-				ItemDisplay.Navigate(InstanceViewModel.FolderSettings.GetLayoutType(FilesystemViewModel.WorkingDirectory), args);
+				ItemDisplay.Navigate(layout, args);
```
Comment (hishitetsu) [inline src/Files.App/Views/Shells/BaseShellPage.cs:501]:
```suggestion
				NavigateToPath(FilesystemViewModel.WorkingDirectory, layout, args);
```
We should always call `NavigateToPath` at page transitions.

Comment (yaira2) [reply src/Files.App/Views/Shells/BaseShellPage.cs:501]: What's the difference?

**Review by hishitetsu — APPROVED** [review]
LGTM

File: src/Files.App/Views/Shells/BaseShellPage.cs (hunk 1 above)
Comment (hishitetsu) [reply src/Files.App/Views/Shells/BaseShellPage.cs]: The processing is specific to the implementation class of IShellPage.
Especially in CloumnShellPage, calling Navigate directly may cause unexpected behavior.

**Pull request merged**